mhkit/utils/stat_utils.py

```diff
@@ -1,9 +1,57 @@
-from mhkit import qc
+"""
+This module contains functions to perform various statistical calculations 
+on continuous data. It includes functions for calculating statistics such as
+mean, max, min, and standard deviation over specific windows, as well as functions
+for vector/directional statistics. The module also provides utility functions 
+to unwrap vectors, compute magnitudes and phases in 2D/3D, and calculate 
+the root mean squared values of vector components.
+
+Functions:
+----------
+- get_statistics: Calculates statistics for continuous data.
+- vector_statistics: Calculates vector mean and standard deviation.
+- unwrap_vector: Unwraps vector data to fall within a 0-360 degree range.
+- magnitude_phase: Computes magnitude and phase for 2D or 3D data.
+- unorm: Computes root mean squared value of 3D vectors.
+"""
+
```

Comment:
module docstrings required by pylint

Reply:
great to see this. This will tie in nicely with my goals in https://github.com/MHKiT-Software/MHKiT/pull/80 and https://github.com/MHKiT-Software/MHKiT-Python/pull/360 too and reduce maintenance of our docs. Do you have a convenient method to quickly produce these doc strings?

also, is it pylint required to list out the module's functions in the module-level docstring? Right now we're using ``sphinx autosummary`` to produce similar content at the top of the each [module's API](https://mhkit-software.github.io/MHKiT/mhkit-python/api.utils.html#utils-module). Autosummary formats nicely and will pull the function docstring directly. I could go either way, just want to note how to change our API docs formatting to match the source docstrings

Reply:
Having a consistent list of functions auto generated would be nice and easier to maintain. I like the idea as a future goal to replace the hardcoded list of functions with something auto generated. pylint does not require this or compile a list automatically. Pylint simply requires that there is a docstring at the module level.